src/backend/parser/cypher_expr.c

```diff
@@ -152,12 +152,15 @@ static Node *transform_cypher_expr_recurse(cypher_parsestate *cpstate,
     case T_NullTest:
     {
         NullTest *n = (NullTest *)expr;
+        NullTest *tranformed_expr = makeNode(NullTest);
```

Comment:
Is there a particular reason for creating a new node as opposed to using the passed node?

Reply:
Actually, what about using the transformed node and just checking to see if it was already transformed?